File: src/main/java/com/thehecklers/kcupservice/CoffeeController.java
package com.thehecklers.kcupservice;

import org.springframework.web.bind.annotation.*;

import java.util.Optional;

@RestController
@RequestMapping("/coffees")
public class CoffeeController {
    private final CoffeeRepository repository;

    public CoffeeController(CoffeeRepository repository) {
        this.repository = repository;
    }

    @GetMapping
    Iterable<Coffee> getAllCoffees() {
        return repository.findAll();
    }

    @GetMapping("/{id}")
    Optional<Coffee> getCoffeeById(@PathVariable String id) {
        return repository.findById(id);
    }

    @GetMapping("/search")
    Optional<Coffee> searchForCoffee(@RequestParam(required = false) String code) {
        if (code == null) {
            return Optional.of(repository.findAll().iterator().next());
        } else {
            return repository.findCoffeeByCode(code);
        }
    }
}

File: src/main/kotlin/com/thehecklers/kcupservice/CoffeeController.kt
package com.thehecklers.kcupservice

import org.springframework.web.bind.annotation.*
import java.util.*

@RestController
@RequestMapping("/coffees")
class CoffeeController(private val repository: CoffeeRepository) {
    @GetMapping
    fun getAllCoffees() = repository.findAll()

    @GetMapping("/{id}")
    fun getCoffeeById(@PathVariable id: String) = repository.findById(id)

    @GetMapping("/search")
    fun searchForCoffee(@RequestParam code: String?) =
        if (code.isNullOrEmpty())
            repository.findAll().first()
        else
            repository.findCoffeeByCode(code)
}
